app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt

```diff
@@ -49,13 +50,18 @@ import org.mozilla.fenix.test.Mockable
  */
 @Mockable
 class BackgroundServices(
-    context: Context,
+    private val context: Context,
     historyStorage: PlacesHistoryStorage,
     bookmarkStorage: PlacesBookmarksStorage
 ) {
     companion object {
         const val CLIENT_ID = "a2270f727f45f648"
-        const val REDIRECT_URL = "https://accounts.firefox.com/oauth/success/$CLIENT_ID"
+
+        fun redirectUrl(context: Context) = if (context.isInExperiment(Experiments.asFeatureWebChannelsDisabled)) {
+            "https://accounts.firefox.com/oauth/success/$CLIENT_ID"
+        } else {
+            "urn:ietf:wg:oauth:2.0:oob:oauth-redirect-webchannel"
```

Comment:
I'm not quite sure, but I thought this was the staging server url

Reply:
A redirectUri tells the oauth server how to send a response to the application. We have two kinds of URIs here: one a real URI, oauth server will redirect the browser to it to deliver the code (as a GET parameter), and another is a "pseudo-uri" that acts as a flag for the server, indicating that oauth code will be delivered "out-of-band". That's what `oob` stands for in the URI. Most common "out-of-band" mechanism would be simply presenting user with a code, asking them to copy/paste it to complete the flow. You'd normally see this as `urn:ietf:wg:oauth:2.0:oob` in other contexts (e.g. microsoft and google's oauth2 docs). This isn't actually part of the OAuth2 spec, afaik.

In our case, we have a special snowflake uri: `urn:ietf:wg:oauth:2.0:oob:oauth-redirect-webchannel`, which indicates to the oauth server that code should be delivered via a webchannel protocol.

In practice, both of these URIs really achieve the same thing, but via different means: delivering oauth code to the application. In one case we fish it out from the final redirect URI, in another case we have an explicit communication protocol (via the "webchanel"). The latter is a much richer and more formalized expression of the same idea.